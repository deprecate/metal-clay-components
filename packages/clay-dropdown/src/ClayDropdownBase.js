import 'clay-button';
import 'clay-checkbox';
import 'clay-icon';
import 'clay-radio';
import Dropdown from 'metal-dropdown';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayDropdownBase.soy.js';

/**
 * Implementation of the base for Metal Clay Dropdown.
 */
class ClayDropdownBase extends Dropdown {
  /**
   * Handles Search in Dropdown.
   * @param {!Event} event
   * @protected
   */
  _handleSearch(event) {
    let searchValue = event.delegateTarget.value.toLowerCase();

    if (!this._originalItems) {
      this._originalItems = this.items;
    }

    if (searchValue !== '') {
      this.items = this._originalItems.filter(item => {
        return (
          item.label &&
          item.type !== 'separator' &&
          item.type !== 'header' &&
          item.label.toLowerCase().indexOf(searchValue) !== -1
        );
      });
    } else {
      this.items = this._originalItems;
    }
  }
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayDropdownBase.STATE = {
  /**
   * Element selector used to position dropdown according to trigger position.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default .dropdown-toggle
   */
  alignElementSelector: Config.string()
    .value('.dropdown-toggle')
    .internal(),

  /**
   * Button configuration to place a button at dropdown footer.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default .dropdown-toggle
   */
  button: Config.shapeOf({
    label: Config.string().required(),
    style: Config.oneOf(['primary', 'secondary']).value('primary'),
  }),

  /**
   * Caption text of the dropdown.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default undefined
   */
  caption: Config.string(),

  /**
   * Flag to indicate if menu is expanded.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?bool}
   * @default false
   */
  expanded: Config.bool().value(false),

  /**
   * Help text to be shown on top of the open dropdown.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default undefined
   */
  helpText: Config.string(),

  /**
   * Position in which item indicator symbols will be placed. Needed if any
   * item has indicators.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default undefined
   */
  indicatorsPosition: Config.oneOf(['left', 'right']),

  /**
   * List of menu items.
   * @instance
   * @memberof ClayDropdownBase
   * @type {!Array}
   * @default undefined
   */
  items: Config.arrayOf(
    Config.shapeOf({
      active: Config.bool().value(false),
      checked: Config.bool().value(false),
      disabled: Config.bool().value(false),
      indicatorSymbol: Config.string(),
      inputName: Config.string(),
      inputValue: Config.string(),
      label: Config.string(),
      type: Config.oneOf([
        'checkbox',
        'header',
        'item',
        'radio',
        'separator',
      ]).value('item'),
      url: Config.string().required(),
    })
  ).required(),

  /**
   * Flag to indicate if menu has a search field and search through elements
   * is possible.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?bool}
   * @default false
   */
  searchable: Config.bool().value(false),

  /**
   * The path to the SVG spritemap file containing the icons.
   * @instance
   * @memberof ClayDropdownBase
   * @type {?string}
   * @default undefined
   */
  spritemap: Config.string(),

  /**
   * Label of the trigger button.
   * @instance
   * @memberof ClayDropdownBase
   * @type {!string}
   * @default undefined
   */
  triggerLabel: Config.any().required(),

  /**
   * Style of the trigger button.
   * @instance
   * @memberof ClayDropdownBase
   * @type {!string}
   * @default unstyled
   */
  triggerStyle: Config.oneOf([
    'borderless',
    'link',
    'primary',
    'secondary',
    'unstyled',
  ]).value('unstyled'),
};

Soy.register(ClayDropdownBase, templates);

export {ClayDropdownBase};
export default ClayDropdownBase;
