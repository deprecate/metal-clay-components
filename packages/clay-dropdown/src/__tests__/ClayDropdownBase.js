import ClayDropdownBase from '../ClayDropdownBase';

let clayDropdownBase;

describe('ClayDropdownBase', function() {
  afterEach(() => {
    if (clayDropdownBase) {
      clayDropdownBase.dispose();
    }
  });

  it('should render a dropdown with items of type item', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
        {
          label: 'Item 2',
          url: 'item2url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type item and active', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          active: true,
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type item and disabled', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          disabled: true,
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type checkbox', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          inputName: 'checkbox1name',
          inputValue: 'checkbox1value',
          label: 'Item 1',
          type: 'checkbox',
        },
        {
          inputName: 'checkbox2name',
          inputValue: 'checkbox2value',
          label: 'Item 2',
          type: 'checkbox',
        },
      ],
      triggerLabel: 'Trigger',
      type: 'form',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type checkbox and checked', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          checked: true,
          inputName: 'checkbox1name',
          inputValue: 'checkbox1value',
          label: 'Item 1',
          type: 'checkbox',
        },
      ],
      triggerLabel: 'Trigger',
      type: 'form',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  // eslint-disable-next-line
  it('should render a dropdown with items of type checkbox and disabled', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          disabled: true,
          inputName: 'checkbox1name',
          inputValue: 'checkbox1value',
          label: 'Item 1',
          type: 'checkbox',
        },
      ],
      triggerLabel: 'Trigger',
      type: 'form',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type radio', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          items: [
            {
              label: 'Item 1',
              inputValue: '1',
            },
            {
              label: 'Item 2',
              inputValue: '2',
            },
          ],
          inputName: 'item1radio',
          label: 'Group 1',
          separator: true,
          type: 'radiogroup',
        },
        {
          items: [
            {
              label: 'Item 4',
              inputValue: '4',
            },
            {
              label: 'Item 5',
              inputValue: '5',
            },
          ],
          inputName: 'item2radio',
          label: 'Group 2',
          type: 'radiogroup',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger Label',
      type: 'form',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type radio and checked', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          items: [
            {
              label: 'Item 1',
              inputValue: '1',
            },
            {
              checked: true,
              label: 'Item 2',
              inputValue: '2',
            },
          ],
          inputName: 'item1radio',
          label: 'Group 1',
          separator: true,
          type: 'radiogroup',
        },
        {
          items: [
            {
              label: 'Item 4',
              inputValue: '4',
            },
            {
              checked: true,
              label: 'Item 5',
              inputValue: '5',
            },
          ],
          inputName: 'item2radio',
          label: 'Group 2',
          type: 'radiogroup',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger Label',
      type: 'form',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items of type radio and disabled', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          items: [
            {
              label: 'Item 1',
              inputValue: '1',
            },
            {
              disabled: true,
              label: 'Item 3',
              inputValue: '3',
            },
          ],
          inputName: 'item1radio',
          label: 'Group 1',
          separator: true,
          type: 'radiogroup',
        },
        {
          items: [
            {
              label: 'Item 4',
              inputValue: '4',
            },
            {
              disabled: true,
              label: 'Item 6',
              inputValue: '6',
            },
          ],
          inputName: 'item2radio',
          label: 'Group 2',
          type: 'radiogroup',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger Label',
      type: 'form',
    });
    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with groups', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          label: 'Group 1',
          items: [
            {
              active: true,
              label: 'Item 1',
              url: '#1',
            },
            {
              label: 'Item 2',
              url: '#2',
            },
          ],
          type: 'group',
        },
        {
          label: 'Group 2',
          items: [
            {
              label: 'Item 3',
              url: '#3',
            },
            {
              disabled: true,
              label: 'Item 3',
              url: '#4',
            },
          ],
          type: 'group',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger Label',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with items with separators', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          label: 'Element 1',
          separator: true,
          url: '#1',
        },
        {
          label: 'Group 1',
          items: [
            {
              label: 'Item 2',
              url: '#2',
            },
            {
              label: 'Item 3',
              url: '#3',
            },
          ],
          separator: true,
          type: 'group',
        },
        {
          label: 'Element 4',
          url: '#4',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger Label',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with indicators on the left', () => {
    clayDropdownBase = new ClayDropdownBase({
      indicatorsPosition: 'left',
      items: [
        {
          indicatorSymbol: 'check',
          label: 'Item 1',
          url: 'item1url',
        },
        {
          indicatorSymbol: 'check',
          label: 'Item 2',
          url: 'item2url',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with indicators on the right', () => {
    clayDropdownBase = new ClayDropdownBase({
      indicatorsPosition: 'right',
      items: [
        {
          indicatorSymbol: 'check',
          label: 'Item 1',
          url: 'item1url',
        },
        {
          indicatorSymbol: 'check',
          label: 'Item 2',
          url: 'item2url',
        },
      ],
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with help text', () => {
    clayDropdownBase = new ClayDropdownBase({
      helpText: 'Help!',
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with primary button', () => {
    clayDropdownBase = new ClayDropdownBase({
      button: {
        label: 'Button',
        style: 'primary',
      },
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with secondary button', () => {
    clayDropdownBase = new ClayDropdownBase({
      button: {
        label: 'Button',
        style: 'secondary',
      },
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a dropdown with caption', () => {
    clayDropdownBase = new ClayDropdownBase({
      caption: 'My caption!',
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
      ],
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should render a searchable dropdown', () => {
    clayDropdownBase = new ClayDropdownBase({
      items: [
        {
          label: 'Item 1',
          url: 'item1url',
        },
        {
          label: 'Item 2',
          url: 'item2url',
        },
      ],
      searchable: true,
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase).toMatchSnapshot();
  });

  it('should filter items', () => {
    let item1 = {
      label: 'Item 1',
      url: 'item1url',
    };

    let item2 = {
      label: 'Item 2',
      url: 'item2url',
    };

    let items = [item1, item2];

    clayDropdownBase = new ClayDropdownBase({
      items: items,
      searchable: true,
      spritemap: 'icons.svg',
      triggerLabel: 'Trigger',
    });

    expect(clayDropdownBase.items).toEqual(items);

    let input = clayDropdownBase.refs['searchInput'];
    input.value = 'Item 1';

    clayDropdownBase._handleSearch({
      delegateTarget: input,
    });

    expect(clayDropdownBase.items).toEqual([item1]);
  });
});
