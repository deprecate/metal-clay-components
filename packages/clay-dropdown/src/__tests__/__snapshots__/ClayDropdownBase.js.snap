// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayDropdownBase should render a dropdown with caption 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div class="dropdown-caption">My caption!</div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with help text 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="alert alert-fluid alert-info" role="alert">Help!</div>
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with indicators on the left 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu dropdown-menu-indicator-start">
    <div>
      <a class="dropdown-item" href="item1url">
        <div class="dropdown-item-indicator">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-check">
            <use xlink:href="icons.svg#check"></use>
          </svg>
        </div>Item 1</a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">
        <div class="dropdown-item-indicator">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-check">
            <use xlink:href="icons.svg#check"></use>
          </svg>
        </div>Item 2</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with indicators on the right 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu dropdown-menu-indicator-end">
    <div>
      <a class="dropdown-item" href="item1url">Item 1
        <div class="dropdown-item-indicator">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-check">
            <use xlink:href="icons.svg#check"></use>
          </svg>
        </div>
      </a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">Item 2
        <div class="dropdown-item-indicator">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-check">
            <use xlink:href="icons.svg#check"></use>
          </svg>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type checkbox 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-checkbox">
        <input class="custom-control-input" name="checkbox1name" value="checkbox1value" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
    <div class="dropdown-item">
      <label class="custom-control custom-checkbox">
        <input class="custom-control-input" name="checkbox2name" value="checkbox2value" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 2</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type checkbox and checked 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-checkbox">
        <input checked="" class="custom-control-input" name="checkbox1name" value="checkbox1value" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type checkbox and disabled 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-checkbox">
        <input disabled="" class="custom-control-input" name="checkbox1name" value="checkbox1value" ref="input" type="checkbox">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type group 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">Item 2</a>
    </div>
    <div class="dropdown-divider"></div>
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">Item 2</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type header 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-subheader">Group 1</div>
    <div class="dropdown-subheader">Group 2</div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type item 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">Item 2</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type item and active 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item active" href="item1url">Item 1</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type item and disabled 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item disabled" href="javascript:;">Item 1</a>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type radio 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input class="custom-control-input" name="radioname" value="radio1value" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input class="custom-control-input" name="radioname" value="radio2value" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 2</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type radio 2`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input class="custom-control-input" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input class="custom-control-input" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 2</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type radio and checked 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input checked="" class="custom-control-input" name="radio1name" value="radio1value" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type radio and disabled 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-item">
      <label class="custom-control custom-radio">
        <input disabled="" class="custom-control-input" name="radio1name" value="radio1value" ref="input" type="radio" role="radio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Item 1</span>
      </label>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with items of type separator 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-divider"></div>
    <div class="dropdown-divider"></div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with primary button 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div class="dropdown-section">
      <button class="btn btn-block btn-primary">Button</button>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a dropdown with secondary button 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div class="dropdown-section">
      <button class="btn btn-block btn-secondary">Button</button>
    </div>
  </div>
</div>
`;

exports[`ClayDropdownBase should render a searchable dropdown 1`] = `
<div class="dropdown">
  <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">Trigger</button>
  <div class="dropdown-menu">
    <div class="dropdown-section">
      <div class="input-group input-group-inset">
        <div class="input-group-input">
          <input aria-label="Search for..." class="form-control" placeholder="Search for..." type="text" ref="searchInput">
        </div>
        <span class="input-group-inset-item">
          <button class="btn btn-link" aria-label="search">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-search">
              <use xlink:href="icons.svg#search"></use>
            </svg>
          </button>
        </span>
      </div>
    </div>
    <div>
      <a class="dropdown-item" href="item1url">Item 1</a>
    </div>
    <div>
      <a class="dropdown-item" href="item2url">Item 2</a>
    </div>
  </div>
</div>
`;
